compiler+runtime/src/cpp/jank/read/lex.cpp

```diff
@@ -782,21 +889,58 @@ namespace jank::read
             return ok(token{ token_start, pos - token_start, token_kind::deref });
           }
         default:
+          if (is_symbol_char(start))
```

Comment:
Why would we have a symbol char in this default? Is it only because it's a utf8 char? If so, let's use that predicate.

Also, please add a comment explaining why we need to handle symbols a second time here.